src/Shared/Microsoft.AspNetCore.Razor.Test.Common/Language/RazorCodeDocumentProcessor.cs

```diff
@@ -24,41 +24,42 @@ public static RazorCodeDocumentProcessor From(RazorProjectEngine projectEngine,
     public RazorCodeDocumentProcessor ExecutePhasesThrough<T>()
         where T : IRazorEnginePhase
     {
-        ProjectEngine.ExecutePhasesThrough<T>(CodeDocument);
+        var codeDocument = ProjectEngine.ExecutePhasesThrough<T>(CodeDocument);
 
-        return this;
+        return From(ProjectEngine, codeDocument);
     }
 
     public RazorCodeDocumentProcessor ExecutePass<T>()
         where T : IntermediateNodePassBase, new()
     {
-        ProjectEngine.ExecutePass<T>(CodeDocument);
+        var codeDocument = ProjectEngine.ExecutePass<T>(CodeDocument);
 
-        return this;
+        return From(ProjectEngine, codeDocument);
     }
 
     public RazorCodeDocumentProcessor ExecutePass<T>(Func<T> passFactory)
         where T : IntermediateNodePassBase
     {
-        ProjectEngine.ExecutePass<T>(CodeDocument, passFactory);
+        var codeDocument = ProjectEngine.ExecutePass<T>(CodeDocument, passFactory);
 
-        return this;
+        return From(ProjectEngine, codeDocument);
     }
 
     public RazorCodeDocumentProcessor ExecutePhase<T>(RazorCodeDocument codeDocument)
         where T : IRazorEnginePhase, new()
     {
-        ProjectEngine.ExecutePhase<T>(codeDocument);
+        codeDocument = ProjectEngine.ExecutePhase<T>(codeDocument);
```

Comment:
>codeDocument [](http://example.com/codeflow?start=8&length=12)

nit: looks like this and the next method aren't used. Since we're breaking public APIs in this PR, can we remove these?